docs/upgrade/2.0-3.0.md

```diff
@@ -0,0 +1,302 @@
+well.
+
+To run the whole test suite (client and server), run `make test`. You can also
+individually test just client or just server using `make test-client` and `make
+test-server` respectively.
+
+### Provision and deploy w/ chef and cap
+
+A major goal of Madison 3.0 is to ease the process of standing up and managing
+your own instance of Madison. The customization work covered above was one piece
+of this. Another piece is the automated server provisioning using
+[Chef](https://www.chef.io/) and easier deployment using
+[Capistrano](http://capistranorb.com)
+
+Instead of covering everything involved in the process here, we've created a
+document with instructions on setting up, provisioning, and deploying to a
+server using these tools in `docs/DEPLOY.md`.
+
+## Suggested upgrade path
+
+What follows are some steps you can follow to upgrade an existing Madison
+instance from it's current version to the 3.0 version. Because there are some
+major changes to how the project is structured, you will want to take extra care
+to ensure that everything works accordingly.
+
+### Backup current codebase & database
+
+Make sure to create a backup copy of your current codebase, and a dump of your
+current database, should anything go awry.
+
+To copy your codebase, simply running the following command should be enough:
+
+```
+$ cp -r /path/to/madison/codebase /path/to/madison/codebase.backup
+```
+
+To backup your database, assuming you're using MySQL, use the following:
+
+```
+$ mysqldump --user=username --password=password database_name > madison_db_dump.sql
+```
+
+Make sure to have access to the DB dump file and keep it in a safe place. If
+something goes wrong, you can simply import the database dump. It will restore
+it to it's previous state. To restore, use this command:
+
+```
+$ mysql -u username -p password database_name < madison_db_dump.sql
+```
+
+### Pull down Madison 3.0 code from GitHub!
+
+You will want to pull in the changes for Madison 3.0 from our main GitHub repo,
+located at https://github.com/opengovfoundation/madison. If your project is not
+set up to track from the primary repo, you will want to add it as a remote:
+
+```
+$ git remote add opengovfoundation git@github.com:opengovfoundation/madison.git
+```
+
+Now you should be able to do:
+
+```
+$ git pull opengovfoundation master
+```
+
+This will pull down the master branch to your local instance. If you're doing
+this in production *please make sure you have your codebase backed up as
+mentioned above*!
+
+This should put the new folder structure in place with your local project. To
+confirm, make sure the top level directory has `client` and `server` folders,
+and **does not** have a top level  `public` directory.
+
+### Configuration
+
+Previously, the primary configuration file is `.env` in the top level directory.
+That file now resides in the `server` folder. You will want to move it to be in
+the `server` folder and also check `server/.env.example` to make sure all
```

Comment:
Might want to mention that the `APP_KEY` needs have been changed and to ensure there are at least 32 characters in it. Also that they pull over the `QUEUE_DRIVER` setting from the example and set the `APP_URL` variable. I think that is all the new and changed variables.